cli/internal/cmd/upgradecheck.go

```diff
@@ -510,6 +570,81 @@ func getCompatibleImageMeasurements(ctx context.Context, writer io.Writer, clien
 	return upgrades, nil
 }
 
+type versionFetcher interface {
+	FetchVersionList(ctx context.Context, list versionsapi.List) (versionsapi.List, error)
+	FetchCLIInfo(ctx context.Context, cliInfo versionsapi.CLIInfo) (versionsapi.CLIInfo, error)
+}
+
+// newCLIVersions returns a list of versions of the CLI which are a valid upgrade.
+func (v *versionCollector) newCLIVersions(ctx context.Context, currentKubernetesVersion string) ([]string, error) {
+	list := versionsapi.List{
+		Ref:         v.flags.ref,
+		Stream:      v.flags.stream,
+		Granularity: versionsapi.GranularityMajor,
+		Base:        "v2",
```

Comment:
Probably not a problem in the near future, but this should likely be based on the current major version.
```suggestion
func (v *versionCollector) newCLIVersions(ctx context.Context) ([]string, error) {
	cliVersion, err := conSemver.New(constants.VersionInfo())
	if err != nil {
		return nil, fmt.Errorf("parsing current CLI version: %w", err)
	}
	list := versionsapi.List{
		Ref:         v.flags.ref,
		Stream:      v.flags.stream,
		Granularity: versionsapi.GranularityMajor,
		Base:        fmt.Sprintf("v%d", cliVersion.Major),
```